lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -348,7 +350,17 @@ def _on_start(self, event: StartEvent):
 
         # request the start of OpenSearch
         self.status.set(WaitingStatus(RequestUnitServiceOps.format("start")))
-        self._start_opensearch_event.emit()
+
+        deployment_desc = self.opensearch_peer_cm.deployment_desc()
+        ignore_lock = (
+            "data" not in deployment_desc.config.roles
+            and deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            or "data" in deployment_desc.config.roles
+            and self.unit.is_leader()
+            and deployment_desc.typ == DeploymentType.OTHER
+            and not self.peers_data.get(Scope.APP, "security_index_initialised", False)
+        )
```

Comment:
I think we should block / defer the start until after the first data node joins (we'll get notified through the large deployment relations)

Reply:
adjusted.